profile.d/00-pengwin.sh

```diff
@@ -1,5 +1,22 @@
-# enable external x display
-export DISPLAY=:0
+# check whether it is WSL1 for WSL2 by using gcc version kernel built with
+# check whether we have wsl.exe in path
+if ( which wsl.exe >/dev/null ) && ( $(which wsl.exe) -l -v >/dev/null ) && [ $(wsl.exe -l -v 2>&1 |sed -e "s|\r||g" -e "s|\x00||g" | grep WLinux | awk '{print $3}') -eq 2]; then
+  # enable external x display for WSL 2
+
+  ipconfig_exec=$(wslpath "C:\\Windows\\System32\\ipconfig.exe")
+  if ( which ipconfig.exe >/dev/null ); then
+    ipconfig_exec=$(which ipconfig.exe)
+  fi
+
+  wsl2_d_tmp="eval "$ipconfig_exec" | grep -n WSL | cut -d : -f 1)"
+  wsl2_d_tmp="eval "$ipconfig_exec" | sed ''"$(expr $wsl2_d_tmp + 0)"','"$(expr $wsl2_d_tmp + 4)"'!d' | grep IPv4 | cut -d : -f 2 | sed -e "s|\s||g" -e "s|\r||g")"
+  export DISPLAY=$wsl2_d_tmp:0.0
+  sed -i 's|DISPLAY=.*$|DISPLAY='$wsl2_d_tmp':0\.0|g' /etc/environment
```

Comment:
This script doesn't run as root so this line will fail. Probably it is not necessary. If it is necessary we can add something to the sudoers for this case

Reply:
Yeah, possibly it is not necessary but `/etc/environment` seems to have an issue to change when switch between WSL1 and WSL2. What do you think it's the best way?

Reply:
We can make a script which will update /etc/environment only if it is necessary and put in the sudoers

Reply:
Okay, so I think I will try to remove this part first.

Reply:
Yes let's create a separate task for updating environment, because we need changes in LIBGL_ALWAYS_INDIRECT also in the case that mesa-utils is installed